server/publicapi/items/service.py

```diff
@@ -67,19 +62,169 @@ def get(self, req, lookup):
+        :param whitelist: iterable containing the names of allowed parameters.
+        :param bool allow_filtering: whether or not the filtering parameter is
+            allowed (True by default). Used for disallowing it when retrieving
+            a single object.
+
+        :raises BadParameterValueError: if the request contains more than one
+            value for any of the parameters
+        :raises UnexpectedParameterError: if the request contains a parameter
+            that is not whitelisted
+        """
+        request_params = (request.args or MultiDict())
 
-        for param in request_params.keys():
-            if param not in allowed_params:
+        if not allow_filtering and 'q' in request_params.keys():
+            desc = (
+                "Filtering is not supported when retrieving a single "
+                "object (the \"q\" parameter)"
+            )
+            raise UnexpectedParameterError(desc=desc)
+
+        for param_name in request_params.keys():
+            if param_name not in whitelist:
                 raise UnexpectedParameterError(
-                    desc="Unexpected parameter ({})".format(param)
+                    desc="Unexpected parameter ({})".format(param_name)
                 )
 
-        if 'q' in request_params:
-            req.where = request_params['q']
+            if len(request_params.getlist(param_name)) > 1:
+                desc = "Multiple values received for parameter ({})"
+                raise BadParameterValueError(desc=desc.format(param_name))
 
-        return super().get(req, lookup)
+    def _get_date_range(self, request_params):
+        """Extract the start and end date limits from request parameters.
+
+        If start and/or end date parameter is not present, a default value is
+        returned for the missing parameter(s).
+
+        :param dict request_params: request parameter names and their
+            corresponding values
+
+        :return: a (start_date, end_date) tuple with both values being
+            instances of Python's datetime.date
+
+        :raises BadParameterValueError:
+            * if any of the dates is not in the ISO 8601 format
+            * if any of the dates is set in the future
+            * if the start date is bigger than the end date
+        """
+        try:
+            start_date = self._parse_iso_date(request_params.get('start_date'))
+        except ValueError:
+            raise BadParameterValueError(
+                desc="Invalid parameter value (start_date)") from None
+
+        try:
+            end_date = self._parse_iso_date(request_params.get('end_date'))
+        except ValueError:
+            raise BadParameterValueError(
+                desc="Invalid parameter value (end_date)") from None
+
+        # disallow dates in the future
+        future_err_msg = (
+            "{} date ({}) must not be set in the future "
+            "(current server date: {})")
+        today = date.today()
```

Comment:
I assume yes, but it's true that this was never specified exactly. I'm thinking it would probably be better to use the current UTC time, what do you think?